misc/python/materialize/data_ingest/workload.py

```diff
@@ -0,0 +1,147 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+import random
+import time
+from copy import deepcopy
+from typing import Any, Dict, Iterator, List
+
+import pg8000
+
+from materialize.data_ingest.data_type import (
+    DoubleType,
+    FloatType,
+    IntType,
+    LongType,
+    StringType,
+)
+from materialize.data_ingest.definition import (
+    Definition,
+    Delete,
+    Insert,
+    Keyspace,
+    Records,
+    RecordSize,
+    Upsert,
+)
+from materialize.data_ingest.executor import PgExecutor, PrintExecutor
+from materialize.data_ingest.field import Field
+from materialize.data_ingest.transaction import Transaction
+
+
+class Workload:
+    cycle: List[Definition]
+    num_cycles: int
+
+    def generate(self, fields: List[Field]) -> Iterator[Transaction]:
+        for i in range(self.num_cycles):
+            cycle = deepcopy(self.cycle)
+            for definition in cycle:
+                for transaction in definition.generate(fields):
+                    yield transaction
+
+
+class SingleSensorUpdating(Workload):
+    def __init__(self, scale_factor: int) -> None:
+        self.num_cycles = 10 * scale_factor
+        self.cycle: List[Definition] = [
+            Upsert(
+                keyspace=Keyspace.SINGLE_VALUE,
+                count=Records.ONE,
+                record_size=RecordSize.SMALL,
+            )
+        ]
+
+
+class DeleteDataAtEndOfDay(Workload):
+    def __init__(self, scale_factor: int) -> None:
+        self.num_cycles = scale_factor
```

Comment:
So for some workloads `scale_factor` is used verbatim, but for others it is multiplied by 10

If the entire concept of `scale_factor` becomes untenable, consider just allowing a `--max-duration` argument to mzcompose that is an Interval and then run the workload for that long amount of time. At the end, print the number of `cycles/transactions/rowsets` that were actually achieved so that the user can check if the values are acceptable for the desired purpose.